app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -126,60 +141,68 @@ class InfectionMessengerRepositoryImpl(
         temporaryExposureKeysDao.insertSentTemporaryExposureKeys(temporaryExposureKeys)
     }
 
-    override suspend fun fetchDecryptAndStoreNewMessages() {
+    override suspend fun processKeysBasedOnToken(token: String){
+        //we´re processing the batches
+        if (lastScheduledToken != token){
+            Timber.e(IllegalArgumentException("processing of token ${token} when we expect to process ${lastScheduledToken}"))
+        }
+
+        if (token.startsWith(BATCH)){
+            processKeysBasedOnBatchToken(token)
+        } else {
+            processDailyTokens(token)
+        }
+    }
+
+    private fun processDailyTokens(token: String) {
+        //we can assume you are WarningType.GREEN as only in this case we process the tokens
+    }
+
+    private suspend fun processKeysBasedOnBatchToken(token: String) {
+        val warningType = WarningType.RED
+
+        val summary = exposureNotificationRepository.determineRiskWithoutInformingUser(token)
+        val configuration = configurationRepository.observeConfiguration().blockingFirst()
```

Comment:
use `getConfiguration` instead.

Reply:
@falkorichter

Reply:
configurationRepository.getConfiguration() is an optional, how can this be null? we must handle it together